src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,54 @@
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    private static final String DELIMITER_REGEX = ",|:";
+    private static final String CUSTOM_DELIMITER_IDENTIFIER_REGEX = "//(.)\n(.*)";
+    public static final String EXCEPTION_MESSAGE_FOR_EXIST_NEGATIVE_NUMBER = "음수가 존재합니다.";
+
+
+    public static int splitAndSum(String text) {
+        if (isEmpty(text)) {
+            return 0;
+        }
+        if (isCustomDelimiter(text)) {
+            return sum(getTokensByCustomDelimiter(text));
+        }
+        return sum(text.split(DELIMITER_REGEX));
+    }
+
+    private static boolean isEmpty(String text) {
+        return text == null || text.isEmpty();
+    }
+
+    private static boolean isCustomDelimiter(String text) {
+        Matcher m = Pattern.compile(CUSTOM_DELIMITER_IDENTIFIER_REGEX).matcher(text);
+        return m.find();
+    }
+
+    private static String[] getTokensByCustomDelimiter(String text) {
+        Matcher m = Pattern.compile(CUSTOM_DELIMITER_IDENTIFIER_REGEX).matcher(text);
```

Comment:
현재 메서드 호출할때마다 Pattern.compile()을 호출하고 있네요 :)
Pattern.compile 내부 구현을 보면
```
// Pattern
public static Pattern compile(String regex) {
    return new Pattern(regex, 0);
}
```

매번 메서드 호출시 성능이 비싼 Pattern 인스턴스를 생성하고 있습니다.
Pattern 인스턴스는 한번 쓰고 버려지기 때문에 성능적으로 좋지 않는데요 :)
자바 doc에서도 한번 컴파일 후 재사용 할것을 권장하고 있습니다!

```
If a pattern is to be used multiple times, compiling it once and reusing
it will be more efficient than invoking this method each time
```
 
불필요한 객체 생성을 피하도록 개선해주시면 좋을 것 같아요 😄